### PR TITLE
Example App Inception

### PR DESCRIPTION
**Pull request by ollie-stytch**

### CODE REVIEW

File: .env.template [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# The below values may be found in your Stytch Dashboard: https://stytch.com/dashboard/api-keys
+NEXT_PUBLIC_STYTCH_PUBLIC_TOKEN=YOUR_PUBLIC_TOKEN_HERE
+NEXT_PUBLIC_TEST_API_URL=https://test.stytch.com
+NEXT_PUBLIC_SDK_BACKEND_URL=https://web.stytch.com
+NEXT_PUBLIC_LIVE_API_URL=https://api.stytch.com
+NEXT_PUBLIC_DFP_BACKEND_URL=https://telemetry.stytch.com
+NEXT_PUBLIC_CLIENTSIDE_SERVICES_IFRAME_URL=https://js.stytch.com/clientside-services/index.html
```
Comment (ollie-stytch) [inline .env.template]: Add secret and id

File: lib/api.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const deleteUser = async (member_id: string) =>
+  fetch('/api/delete_user', {
+    method: 'POST',
+    body: JSON.stringify({
+      member_id,
```
Comment (max-stytch) [inline lib/api.ts]: member_id?

File: pages/_app.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import { ThemeProvider, createTheme } from '@mui/material/styles';
+import { Box } from '@mui/material';
+import { StytchClientOptions } from '@stytch/vanilla-js';
+
+const stytchClientOptions = {
+  cookieOptions: {
+    jwtCookieName: `stytch_session_passkey_demo_jwt`,
+    opaqueTokenCookieName: `stytch_session_passkey_demo`,
+  },
+  endpoints: {
```
Comment (max-stytch) [inline pages/_app.tsx]: None of this `enpoint` obj is needed

File: pages/dashboard.tsx [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    } else {
+      stytch.otps.email
+        .send(user?.emails?.at(0).email, {
+          expiration_minutes: 5,
+        })
+        .then((resp) => {
+          setMethodID(resp.method_id);
+        })
+        .catch(() => {
+          setErr('Error occurred! Check console log.');
```
Comment (max-stytch) [inline pages/dashboard.tsx]: Should we log the error here, then?

File: pages/dashboard.tsx [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+    if (type === StepUpType.phone) {
+      stytch.otps.sms
+        .send(user?.phone_numbers?.at(0).phone_number, {
+          expiration_minutes: 5,
+        })
+        .then((resp) => {
+          setMethodID(resp.method_id);
+        })
+        .catch((e) => {
+          setErr(e);
```
Comment (max-stytch) [inline pages/dashboard.tsx]: why is err handling different here

File: pages/dashboard.tsx [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+
+  return (
+    <>
+      {displayRegisterPasskey ? (
+        <StytchPasskeyRegistration
+          config={{
+            products: [Products.emailMagicLinks],
+            emailMagicLinksOptions: {
+              loginRedirectURL: 'http://localhost:3000/?type=eml',
+              signupRedirectURL: 'http://localhost:3000/?type=eml',
```
Comment (max-stytch) [inline pages/dashboard.tsx]: Do we need any products? Isn't user guaranteed to be logged in?

File: pages/dashboard.tsx [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+    ((sessionHasPhoneFactor && !sessionHasWebauthnFactor) || (sessionHasEmailFactor && !sessionHasWebauthnFactor)) &&
+    user?.webauthn_registrations?.length > 0;
+  const shouldPromptPhone = !sessionHasPhoneFactor && sessionHasWebauthnFactor && user?.phone_numbers?.length > 0;
+  const shouldPromptEmail = !sessionHasEmailFactor && sessionHasWebauthnFactor && user?.emails?.length > 0;
+
+  useEffect(() => {
+    if (isInitialized && user === null) {
+      router.push('/login');
+    } else if (session) {
+      session.authentication_factors.forEach((factor: AuthenticationFactor) => {
```
Comment (max-stytch) [inline pages/dashboard.tsx]: these shouldn't be in the state - they can be calculated directly from useStytchSession

File: pages/login.tsx [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+  const isMobile = useMediaQuery((theme: any) => theme.breakpoints.down('md'));
+
+  return (
+    <>
+      <Head>
+        <title>Survey Amp | Login</title>
+      </Head>
+      <Box width={'100%'} height="100vh" display="flex">
+        <Box display="flex" flexDirection={'column'} flexGrow={1} minWidth={isMobile ? undefined : 475}>
+          {/* Header */}
```
Comment (max-stytch) [inline pages/login.tsx]: wat dis

File: pages/dashboard.tsx (hunk 7 above)
Comment (max-stytch) [reply pages/dashboard.tsx]: e.g. 
```
const sessionHasPhoneFactor = !!session?.authentication_factors.find(factor => factor.delivery_method === 'sms')
```

File: components/SideNavBar.tsx [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+import Search from '@mui/icons-material/Search';
+import LogoutIcon from '@mui/icons-material/Logout';
+
+type Props = {
+  logout: () => void;
+};
+
+function SideNavBar(props: Props) {
+  return (
+    <Box display={'flex'} minWidth="300px" padding={3} flexDirection="column">
```
Comment (anmol-stytch) [inline components/SideNavBar.tsx:14]: nit don't need `{}`

File: components/SideNavBar.tsx [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+                <Search />
+              </InputAdornment>
+            ),
+          }}
+          variant="outlined"
+          placeholder="Search..."
+        />
+      </Box>
+      <Stack marginTop={6} gap={2} flexGrow={1}>
+        <Box display={'flex'} alignItems="center">
```
Comment (anmol-stytch) [inline components/SideNavBar.tsx:31]: same

File: components/mobile/ResetUserStateButton.tsx [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+        <>
+            <Button variant="contained" color="error" onClick={handleClickOpen}>Reset User State</Button>
+            <Dialog
+                open={open}
+                onClose={handleClose}
+                aria-labelledby="alert-dialog-title"
+                aria-describedby="alert-dialog-description"
+            >
+                <DialogTitle id="alert-dialog-title">
+                    {"Delete this user"}
```
Comment (anmol-stytch) [inline components/mobile/ResetUserStateButton.tsx]: Can just be `Delete this user`

File: lib/userService.ts [hunk 12]
```diff
@@ -0,0 +1,10 @@
+import loadStytch from './stytchClient';
+
+const stytch = loadStytch();
+
+export const UserService = {
+  // TODO: get rid of this functionality and get rid of delete user registrations button
```
Comment (anmol-stytch) [inline lib/userService.ts]: Is this a now TODO or a future one?

File: pages/_app.tsx [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    caption: {
+      fontSize: 16,
+      lineHeight: '20px',
+    },
+    body1: {
+      fontSize: 18,
+      lineHeight: '25px',
+    },
+  },
+});
```
Comment (anmol-stytch) [inline pages/_app.tsx]: nit you could move this to a theme.ts file

File: pages/api/delete_user.ts [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+
+async function handler(req: NextApiRequest, res: NextApiResponse) {
+  try {
+    const { user_id } = JSON.parse(req.body);
+    await UserService.delete(user_id);
+    return res.status(200).end();
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.log(e);
+    return res.status(400).end();
```
Comment (anmol-stytch) [inline pages/api/delete_user.ts:12]: very small nit: do you want to send any JSON with these responses?

File: pages/dashboard.tsx [hunk 15]
```diff
@@ -0,0 +1,221 @@
...
+  email = 'email',
+  phone = 'phone',
+  webauthn = 'webauthn',
+}
+
+const StepUp = ({ type }: { type: StepUpType }) => {
+  const [inputValue, setInputValue] = useState('');
+  const [methodID, setMethodID] = useState('');
+  const { user } = useStytchUser();
+  const [err, setErr] = useState('');
```
Comment (anmol-stytch) [inline pages/dashboard.tsx]: nit: prefer error and setError

File: pages/dashboard.tsx [hunk 16]
```diff
@@ -0,0 +1,221 @@
...
+  const stytch = useStytch();
+
+  const validateOTPButtonClick = () => {
+    stytch.otps.authenticate(inputValue, methodID, {
+      session_duration_minutes: 60,
```
Comment (anmol-stytch) [inline pages/dashboard.tsx]: do you want to extract `60` as a constant?

File: pages/dashboard.tsx [hunk 17]
```diff
@@ -0,0 +1,221 @@
...
+  };
+
+  return (
+    <>
+      <Head>
+        <title>Survey Amp | Dashboard</title>
+      </Head>
+      <Box display={'flex'} width="100%" height="100vh" flexDirection={'column'}>
+        {isMobile ? (
+          <MobileHeader logout={() => logout()} />
```
Comment (anmol-stytch) [inline pages/dashboard.tsx]: nit: logout={logout}

**Review by anmol-stytch — APPROVED** [review]

**Pull request merged**